### DISCUSSION TITLE
TypeError: "r.split is not a function" when testing a component using react-hook-form

### DISCUSSION THREAD

**Discussion by Dark-Raine** [react-hook-form/react-hook-form#5727] [Q&A]
Currently when trying to use the react testing library or jest with enzyme I get this error for watch() and register()

For context I am manually controlling the fields getting the field values through watch() and using setValue as a callback for my custom fields that triggers on change.

What could possibly be going wrong? The app runs fine only issue is when i try to render it in testing.
Would be very appreciative, thanks![image](https://user-images.githubusercontent.com/22915704/123320309-7b10b500-d529-11eb-9f5f-519e9819af7b.png)

**Comment by bluebill1049** [comment]
what's in the `field` object?

**Reply by Dark-Raine** [reply]
Hey bill, this is the object,
```
    {
      "id": 1,
      "title": "first field",
      "type": "single_line",
      "input_guidance": "enter something",
      "value": "something",
      "validations": {
        "required": true,
        "character_limit": null,
        "character_restriction": null,
        "input_format": null
      },
      "options": "[]"
    }

**Reply by bluebill1049** [reply]
`register(1, ...) ` i don't think that's going to work with register api

**Reply by Dark-Raine** [reply]
Wow I feel so silly for not realising it's because I was passing an integer and not a string to register's first argument. Thanks for helping me realise!

**Reply by bluebill1049** [reply]
No worries.
